### PR TITLE
[KYUUBI #1866][DOCS] Add flink sql engine quick start

### PR DESCRIPTION
**Pull request by deadwind4**
### _Why are the changes needed?_

Add quick start documents of the Flink SQL Engine.

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by deadwind4** [comment]
The PR is a subtask of #1866. If merge this PR, don't close #1866.

File: docs/quick_start/quick_start.md [hunk 1]
```diff
@@ -42,7 +42,8 @@ The Kyuubi release package you downloaded or built contains the rest prerequisit
 Components| Role | Optional | Version | Remarks
 --- | --- | --- | --- | ---
 Java | Java<br>Runtime<br>Environment | Required | Java 8/11 | Kyuubi is pre-built with Java 8
-Spark | Distributed<br>SQL<br>Engine | Required | 3.0.0 and above | By default Kyuubi binary release is delivered without<br> a Spark tarball.
+Spark | Distributed<br>SQL<br>Engine | Optional | 3.0.0 and above | By default Kyuubi binary release is delivered without<br> a Spark tarball.
+Flink | Distributed<br>SQL<br>Engine | Optional | 1.14.x | By default Kyuubi binary release is delivered without<br> a Flink tarball.
```
Comment (SteNicholas) [inline docs/quick_start/quick_start.md:46]:
```suggestion
Flink | Distributed<br>SQL<br>Engine | Optional | 1.15.0 and above | By default Kyuubi binary release is delivered without<br> a Flink tarball.
```

File: docs/deployment/flink/quick_start.md [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+```
+As shown in the above case, you can retrieve all the operation logs, the result schema, and the result to your client-side in the beeline console.
+
+PS: If you use yarn-session mode, `SELECT * FROM T;` will not get the result yet. This function will come soon。
+
+Additionally, some useful information about the background Flink SQL application associated with this connection is also printed in the operation log. For example, you can get the Flink web UI from the log for debugging or tuning.
+
+![](../../imgs/flink/flink_jobs_page.png)
+
+
```
Comment (SteNicholas) [inline docs/deployment/flink/quick_start.md:207]: No need to add the unnecesary lines.

**Review by SteNicholas — CHANGES_REQUESTED** [review]
@deadwind4, the quick start could be move to the quick start page and I left the minor comments. Please take a look.

**Review by yaooqinn — APPROVED** [review]

File: docs/quick_start/quick_start.md [hunk 3]
```diff
@@ -36,49 +36,50 @@ You can get the most recent stable release of Apache Kyuubi here:
 ## Requirements
 
 These are essential components required for Kyuubi to startup.
-For quick start deployment, the only thing you need is `JAVA_HOME` and `SPARK_HOME` being correctly set.
```
Comment (SteNicholas) [inline docs/quick_start/quick_start.md:39]: Why do you remove this line? The `JAVA_HOME` is the only thing uses need.

File: docs/quick_start/quick_start.md [hunk 4]
```diff
@@ -106,7 +107,11 @@ From top to bottom are:
 
 ## Running Kyuubi
 
-As mentioned above, for a quick start deployment, then only you need to be sure is that your java runtime environment and `SPARK_HOME` are correct.
+As mentioned above, for a quick start deployment, then you need to be sure is that your java runtime environment is correct.
```
Comment (SteNicholas) [inline docs/quick_start/quick_start.md:110]:
```suggestion
As mentioned above, for a quick start deployment, then only you need to be sure is that your java runtime environment is correct.
```

**Comment by yaooqinn** [comment]
@deadwind4 we‘d better compress the picture a bit to make the size smaller, thank you

**Comment by SteNicholas** [comment]
@deadwind4, thanks for your update to address above comments. I have pushed a commit [322f](https://github.com/SteNicholas/incubator-kyuubi/commit/1dcef48091352bd21da2bf8baab95dc6cab2322f) based on your changes. Could you please take a look?

**Review by SteNicholas — APPROVED** [review]
LGTM. @yaooqinn, please help to merge.

**Comment by yaooqinn** [comment]
thanks, merged to master and 1.5

**Pull request closed without merging**